pkg/kube/client.go

```diff
@@ -301,7 +305,13 @@ func createPatch(target *resource.Info, current runtime.Object) ([]byte, types.P
 		patch, err := jsonpatch.CreateMergePatch(oldData, newData)
 		return patch, types.MergePatchType, err
 	}
-	patch, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, versionedObject)
+
+	patchMeta, err := strategicpatch.NewPatchMetaFromStruct(versionedObject)
+	if err != nil {
+		return nil, types.StrategicMergePatchType, errors.Wrap(err, "unable to create patch metadata from object")
+	}
+
+	patch, err := strategicpatch.CreateThreeWayMergePatch(oldData, newData, currentData, patchMeta, true)
```

Comment:
Question for anyone who comes to check this PR out: The last boolean is a flag that forces an overwrite of the incoming data. That is what we consider to be a good default behavior (the new chart always wins). However, an alternative to this would be to have the `--force` flag control this boolean, so the default would be if there are any conflicts, it would error out. Is there any reason to have this second option?

Personally, I think it is better to leave this how it is, but want to point it out in case there are any good use cases we haven't considered